web/src/user_card_popover.js

```diff
@@ -0,0 +1,813 @@
+import * as user_status from "./user_status";
+import * as user_status_ui from "./user_status_ui";
+import * as util from "./util";
+
+let $current_user_info_popover_elem;
+let $current_user_info_popover_manage_menu;
+export let $current_message_info_popover_elem;
+let current_user_sidebar_popover;
+let current_user_sidebar_user_id;
+let userlist_placement = "right";
+
+export function hide_all_user_info_popovers() {
+    hide_user_info_popover_manage_menu();
+    hide_message_info_popover();
+    hide_user_sidebar_popover();
+    hide_user_info_popover();
+}
+
+export function hide_user_sidebar_popover() {
+    if (user_sidebar_popped()) {
+        // this hide_* method looks different from all the others since
+        // the presence list may be redrawn. Due to funkiness with jQuery's .data()
+        // this would confuse $.popover("destroy"), which looks at the .data() attached
+        // to a certain element. We thus save off the .data("popover") in the
+        // show_user_sidebar_popover and inject it here before calling destroy.
+        $("#user_presences").data("popover", current_user_sidebar_popover);
+        $("#user_presences").popover("destroy");
+        current_user_sidebar_user_id = undefined;
+        current_user_sidebar_popover = undefined;
+    }
+}
+
+export function hide_message_info_popover() {
+    if (message_info_popped()) {
+        $current_message_info_popover_elem.popover("destroy");
+        $current_message_info_popover_elem = undefined;
+    }
+}
+
+export function hide_user_info_popover() {
+    if (user_info_popped()) {
+        $current_user_info_popover_elem.popover("destroy");
+        $current_user_info_popover_elem = undefined;
+    }
+}
+
+export function hide_user_info_popover_manage_menu() {
+    if ($current_user_info_popover_manage_menu !== undefined) {
+        $current_user_info_popover_manage_menu.popover("destroy");
+        $current_user_info_popover_manage_menu = undefined;
+    }
+}
+
+export const _test_sort_group_members = sort_group_members;
+
+function sort_group_members(members) {
+    return members.sort((a, b) => util.strcmp(a.full_name, b.fullname));
+}
+
+// exporting these functions for testing purposes
+export const _test_fetch_group_members = fetch_group_members;
+
+function fetch_group_members(member_ids) {
+    return member_ids
+        .map((m) => people.maybe_get_user_by_id(m))
+        .filter((m) => m !== undefined)
+        .map((p) => ({
+            ...p,
+            user_circle_class: buddy_data.get_user_circle_class(p.user_id),
+            is_active: people.is_active_user_for_popover(p.user_id),
+            user_last_seen_time_status: buddy_data.user_last_seen_time_status(p.user_id),
+        }));
+}
+
+// element is the target element to pop off of
+// user is the user whose profile to show
+// message is the message containing it, which should be selected
+export function show_user_group_info_popover(element, group, message) {
```

Comment:
I'm pretty sure this is an unrelated popover that just displays details on user groups -- it should be extracted to its own module, not moved along with the user card popover.